src/ffi/mod.rs

```diff
@@ -867,29 +863,90 @@ fn filter_log_message(msg: String) -> Option<String> {
     None
 }
 
-impl log::Log for telio_logger_cb {
-    fn enabled(&self, metadata: &Metadata) -> bool {
-        metadata.level() <= log::max_level()
+/// Visitor for `tracing` events that converts one field with name equal to `field_name`
+/// value to a message string.
+pub struct TraceFieldVisitor<'a> {
+    field_name: &'static str,
+    metadata: &'a tracing::Metadata<'a>,
+    message: String,
+}
+
+impl<'a> tracing::field::Visit for TraceFieldVisitor<'a> {
+    #[track_caller]
+    fn record_debug(&mut self, field: &tracing::field::Field, value: &dyn fmt::Debug) {
+        // For now we're handling only the message field value, because other fields are not yet used.
+        if field.name() == self.field_name {
+            self.message = format!(
+                "{:#?}:{:#?} {:?}",
+                self.metadata.module_path().unwrap_or("unknown module"),
+                self.metadata.line().unwrap_or(0),
+                value,
+            );
+        }
+    }
+}
+
+pub struct TelioTracingSubscriber {
+    callback: telio_logger_cb,
+    max_level: tracing::Level,
+}
+
+impl TelioTracingSubscriber {
+    pub fn new(callback: telio_logger_cb, max_level: tracing::Level) -> Self {
+        TelioTracingSubscriber {
+            callback,
+            max_level,
+        }
+    }
+}
+
+impl Subscriber for TelioTracingSubscriber {
+    fn enabled(&self, metadata: &tracing::Metadata<'_>) -> bool {
+        metadata.level() <= &tracing::level_filters::STATIC_MAX_LEVEL
+            && metadata.level() <= &self.max_level
+    }
+
+    fn new_span(&self, _span: &tracing::span::Attributes<'_>) -> tracing::span::Id {
+        // TODO using a placeholder for now
+        tracing::span::Id::from_u64(1337)
     }
 
-    fn log(&self, record: &Record) {
-        if !self.enabled(record.metadata()) {
+    fn record(&self, _span: &tracing::span::Id, _values: &tracing::span::Record<'_>) {
+        // TODO
```

Comment:
what should these functions do?

Reply:
These will be implemented when spans are implemented, which is scheduled for a separate task/PR to make the migration a bit simpler.

Reply:
Fair enough, Thanks for explaining.